configure.ac

```diff
@@ -54,17 +54,37 @@ PACKAGE_VERSION_NUM=`printf "0x%02x%02x%02x" "$major" "$minor" "$patch"`
 
 AC_SUBST(PACKAGE_VERSION_NUM)
 
-# Checks for command-line options from nghttp2
-AC_ARG_ENABLE([werror],
-    [AS_HELP_STRING([--enable-werror],
-                    [Turn on compile time warnings])],
-    [werror=$enableval], [werror=no])
+# Checks for command-line options from ngtcp2
+_DEBUGCFLAGS="-O0 -g3"
+_WERRORCFLAGS="-Wall -Werror -Wfatal-errors"
 
 AC_ARG_ENABLE([debug],
     [AS_HELP_STRING([--enable-debug],
                     [Turn on debug output])],
     [debug=$enableval], [debug=no])
 
+if test "x${debug}" = "x${enableval}"; then
+  DEBUGCFLAGS="$_WERRORCFLAGS $_DEBUGCFLAGS"
+  AC_SUBST([DEBUGCFLAGS])
+fi
+
+AC_ARG_ENABLE([werror],
+    [AS_HELP_STRING([--enable-werror],
+                    [Turn on compile time warnings])],
+    [werror=$enableval], [werror=no])
+
+if test "x${werror}" = "x${enableval}"; then
+  WERRORCFLAGS="$_WERRORCFLAGS"
+  if test "x${debug}" = "xno"; then
+# only perform actual substitution if debug is not enabled
+# the following dummy variable declaration is needed to
+# work around an autoconf behaviour: AC_SUBST will not work
+# without a (any) variable declaration in front of it
+    _DUMMY=
+    AC_SUBST([WERRORCFLAGS])
+  fi
+fi
```

Comment:
I think we have already handled werror later in the configure.ac (WARNCFLAGS).  So we don't have to add extra stuff here.